frontend/src/api.tsx

```diff
@@ -0,0 +1,39 @@
+import axios from 'axios';
+
+const windowData = window as any;
+
+export async function login(username: string, password: string): Promise<boolean> {
+  try {
+    const response = await axios({
+      url: '/login',
+      method: 'post',
+      baseURL: 'https://localhost:8080/api/',
+      headers: {
+        CSRF: windowData.csrfToken,
+      },
+      auth: {
+        username,
+        password,
+      },
+    });
+    return response.status === 200;
+  } catch (error) {
+    return false;
+  }
```

Comment:
either return a promise as is and let the caller to handle it, or you can so something like:

```
return { result, error }
```

so it could be used like so:
```ts
const { result, error } = api.login(xxxx)
const { error } = api.login(xxxx)
if (error) {
 ///
}
```